update-tests/src/test/java/io/zeebe/test/ContainerStateExtension.java

```diff
@@ -8,39 +8,52 @@
 package io.zeebe.test;
 
 import io.zeebe.test.util.testcontainers.ManagedVolume;
+import java.util.Optional;
 import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
-import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store;
+import org.junit.jupiter.api.extension.ExtensionContext.Store.CloseableResource;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
 
-public class ContainerStateExtension
-    implements BeforeTestExecutionCallback, AfterTestExecutionCallback {
-  private static final Logger LOG = LoggerFactory.getLogger(ContainerStateExtension.class);
-
-  private final ContainerState state;
-
-  public ContainerStateExtension(final ContainerState state) {
-    this.state = state;
-  }
+/**
+ * This extension injects a new {@link ContainerState} at runtime into any test which adds a {@link
+ * ContainerState} parameter, and stores it in a {@link Store}. This ensures that the resource is
+ * properly closed (since {@link ContainerState} implements {@link CloseableResource}).
+ *
+ * <p>Note however that it currently only supports injecting a single state, since the stored state
```

Comment:
Easily extended if we need more by storing a collection. Note that we'd need to make the collection itself a `CloseableResource` to ensure everything is cleaned up.